src/components/AuthDialog/SignUp/SignUp.tsx

```diff
@@ -0,0 +1,117 @@
+'use client'
+
+import { yupResolver } from '@hookform/resolvers/yup'
+import { SubmitHandler, useForm } from 'react-hook-form'
+
+import { Button } from '@/components/ui/button'
+import {
+  Card,
+  CardContent,
+  CardDescription,
+  CardFooter,
+  CardHeader,
+  CardTitle,
+} from '@/components/ui/card'
+import { Input } from '@/components/ui/input'
+import { signUpSchema } from '@/schemas/authSchemas'
+import { SignUpForm } from '@/types/auth'
+
+interface AuthDialogProps {
+  switchToLogin: () => void
+}
+
+const SignUp: React.FC<AuthDialogProps> = ({ switchToLogin }) => {
+  const signUpMethods = useForm<SignUpForm>({
+    resolver: yupResolver(signUpSchema as any),
+  })
+
+  const onSignUpSubmit: SubmitHandler<SignUpForm> = (data) => {
+    JSON.stringify(data)
+  }
```

Comment:
wrap this in a usecallback

Reply:
fixed [d4c46a4](https://github.com/dark-orga/anitube-website/pull/2/commits/d4c46a4fe8ffbf9e2f2a15e41c88e4b31ea311b6)